org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/ASTConverter.java

```diff
@@ -3638,9 +3638,9 @@ public Pattern convert(org.eclipse.jdt.internal.compiler.ast.TypePattern pattern
 		if(this.ast.apiLevel < AST.JLS22_INTERNAL) {
 			typePattern = new TypePattern(this.ast);
 		} else {
-			if(pattern.local != null && pattern.local.name[0] == '_' && pattern.local.type != null) {
+			if(pattern.local.type != null) {//If there is a type, then SingleVariableDeclaration
```

Comment:
code opt:
replace the if statements with 
typePattern = new TypePattern(this.ast, pattern.local.type != null);

Reply:
Done. Thanks for the wonderful suggestion.